provider/pkg/await/condition/deleted.go

```diff
@@ -141,7 +141,6 @@ func (dc *Deleted) refreshClusterState() {
 	_, err := dc.getter.Get(ctx, dc.Object().GetName(), metav1.GetOptions{})
 	if err == nil {
 		// Still exists.
-		dc.deleted.Store(false)
```

Comment:
Rationale: the deleted field should transition from false to true, never true to false.